### DISCUSSION TITLE
Allow doctype only to be created once (Single Element Type)

### DISCUSSION THREAD

**Discussion by sebastiandammark** [umbraco/Umbraco-CMS#12084] [Features and ideas]
Often there are specific doctypes that should only exist once.
Doctypes like Frontpage, 404, Search, Newslist and such.

It would be awesome if we could get a toggle that would restrict a specific doctype only to exist once on the given location/level.

![single_element_type](https://user-images.githubusercontent.com/1575483/156246713-a6538720-0c67-411a-b6f8-32ec162c297c.png)

**Comment by nathanwoulfe** [comment]
Yup, can absolutely see a use for this. 

Similarly, being able to control which types a particular user/group can create is an issue that comes up for me quite regularly (there's probably an event I can hook, but would be sweet to have in core)

**Comment by patrickdemooij9** [comment]
I once made something for this in v8: https://github.com/umbraco/Umbraco-CMS/pull/9906. This way you could (in code) limit what document types could be created by your user. Unfortunately, it wasn't accepted into v8 in time and I've yet to create it for v9

**Reply by patrickdemooij9** [reply]
Had some time, so I also created a v9 version for it: https://github.com/umbraco/Umbraco-CMS/pull/12116

**Reply by sebastiandammark** [reply]
It looks awesome 👍👍
But I have a hard time seeing the usage if it's code only ?

Heard at the automotive factory 🤣 
**Engineer:** We've decided to add a mega turbo to all of our cars.
**Customer:** How do I activate it, I can't find the button ?
**Engineer:** It's in the engine compartment, so you have to get out of the car every time to activate it.

**Reply by nul800sebastiaan** [reply]
We're not ready to commit to building a UI for this in the CMS just yet (maybe in the future), however, a package could add functionality to configure this from the UI, doctypes can have context apps as well.
